awx/main/scheduler/task_manager.py

```diff
@@ -518,6 +519,19 @@ def process_pending_tasks(self, pending_tasks):
             if not found_acceptable_queue:
                 logger.debug("{} couldn't be scheduled on graph, waiting for next cycle".format(task.log_format))
 
+    def timeout_approval_node(self):
+        workflow_approval = WorkflowApproval.objects.filter(status='pending').prefetch_related('workflow_approval_template')
```

Comment:
I'm tempted to suggest that we could additionally filter these to only return the approval jobs which have timed out. That would avoid double-inspection of most objects on most runs of the task manager, considering that the prior task manager logic already loops over them. But it would involve `F` expressions.